### DISCUSSION TITLE
Inquiry Regarding Merging Two EBSD Files in MTEX

### DISCUSSION THREAD

**Discussion by PengWei97** [mtex-toolbox/mtex#2123] [Ask Anything] [answered]
Dear MTEX Forum Experts,

I hope this message finds you well. I am seeking guidance on a specific task involving EBSD data manipulation in MTEX.

I need to merge two EBSD files based on their overlapping regions, as illustrated below:
![image](https://github.com/mtex-toolbox/mtex/assets/56183775/eff4c665-96d8-43c0-94f2-7bf55419219b)
where the shaded area represents the overlapping part.

Does MTEX offer functionality to merge EBSD files while considering their overlapping areas? If so, could you kindly provide instructions or recommendations on how to achieve this within MTEX?

Any insights, tips, or recommended workflows would be greatly appreciated.

Thank you for your time and assistance.

Best regards,

Wei

**Comment by lfgmorales** [comment, marked as answer]
Hi Wei

Sometime ago Fillipe Ferreira did something like this

<https://github.com/mtex-toolbox/mtex/issues/362>
@EBSD/cat: Remove overlapping data? · Issue #362 · mtex-toolbox/mtex<https://github.com/mtex-toolbox/mtex/issues/362>
github.com<https://github.com/mtex-toolbox/mtex/issues/362>
        [apple-touch-icon-180x180-a80b8e11abe2.png] <https://github.com/mtex-toolbox/mtex/issues/362>


You may have to do some adaptations but that should work

Cheers

Luiz


Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 22 Apr 2024, at 10:02, Peng Wei ***@***.***> wrote:

﻿

Dear MTEX Forum Experts,

I hope this message finds you well. I am seeking guidance on a specific task involving EBSD data manipulation in MTEX.

I need to merge two EBSD files based on their overlapping regions, as illustrated below:
image.png (view on web)<https://github.com/mtex-toolbox/mtex/assets/56183775/eff4c665-96d8-43c0-94f2-7bf55419219b>
where the shaded area represents the overlapping part.

Does MTEX offer functionality to merge EBSD files while considering their overlapping areas? If so, could you kindly provide instructions or recommendations on how to achieve this within MTEX?

Any insights, tips, or recommended workflows would be greatly appreciated.

Thank you for your time and assistance.

Best regards,

Wei

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2123>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN2CBSDQRAXFFOZJLC3Y6S7WTAVCNFSM6AAAAABGSIGYJOVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGU2DCOBTG4>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2123", "url": "https://github.com/mtex-toolbox/mtex/discussions/2123", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by PengWei97** [reply]
Please forgive me for taking so long to reply to you. 

I used a very simple strategy. After determining the overlapping area, I then used `ebsd_t=[ebsd_1, ebsd_2_new];` to complete my purpose.

**Comment by AzdiarGazder** [comment]
Hi @PengWei97,

Please have a look at the "stitch" function located here:
https://github.com/AzdiarGazder/mtexTools/tree/main/stitch

This function does exactly what you're looking for.
Hope this helps.

Warm regards,
Azdi

**Reply by PengWei97** [reply]
Thank you very much for your enthusiastic answer. 

Can the "stitch" function automatically identify overlapping areas?

**Reply by AzdiarGazder** [reply]
Hi @PengWei97,

No - it does not automatically identify overlapping areas.
You have to set that up manually - following which the 2 maps are stitched together as one.
Please have a look at the accompanying demo script and have a read of the function details.

Warm regards,
Azdi

**Reply by PengWei97** [reply]
Ok, I got it. Thank your for sharing.

Warm regards,
Wei
